common-service/src/main/java/org/sunbird/controllers/BaseController.java

```diff
@@ -240,34 +329,180 @@ public static Response createSuccessResponse(String path, Response response, Str
+   * Common exception response handler method.
    *
-   * @param path Request URI path
-   * @return Version of API request
+   * @param e Exception represents the exception occurred while processing the request.
+   * @param request play.mvc.Http.Request
+   * @return Result
    */
-  public static String getApiVersion(String path) {
-    if (StringUtils.isBlank(path)) {
-      ProjectLogger.log("BaseController:getApiVersion: Path is blank.", LoggerEnum.INFO);
-      return "v1";
+  public Result createCommonExceptionResponse(
+      Exception e, Request request, boolean generateTelemetry) {
+
+    ProjectCommonException exception = null;
+    if (e instanceof ProjectCommonException) {
+      exception = (ProjectCommonException) e;
+    } else {
+      exception =
+          new ProjectCommonException(
+              ResponseCode.internalError.getErrorCode(),
+              ResponseCode.internalError.getErrorMessage(),
+              ResponseCode.SERVER_ERROR.getResponseCode());
     }
-    return path.split("[/]")[1];
+
+    Request req = request;
+    if (req == null) {
+      req = request();
+    }
+
+    if (generateTelemetry) {
+      generateTelemetryForExceptionResponse(exception, req);
+    }
+    return createCommonExceptionResult(req.path(), exception, req.method());
   }
 
   /**
-   * Setter method for actor reference or selection object.
+   * Creates a response object in case of an exception.
    *
-   * @param actorRefObj Actor reference or selection object
+   * @param exception Exception which occurred
+   * @return Response object created
    */
-  public static void setActorRef(Object actorRefObj) {
-    actorRef = actorRefObj;
+  private static Response getErrorResponse(Exception exception) {
+    Response response = new Response();
+    ResponseParams resStatus = new ResponseParams();
+    String message = setMessage(exception);
+    resStatus.setErrmsg(message);
+    resStatus.setStatus(StatusType.FAILED.name());
+    if (exception instanceof ProjectCommonException) {
+      ProjectCommonException me = (ProjectCommonException) exception;
+      resStatus.setErr(me.getCode());
+      response.setResponseCode(ResponseCode.getHeaderResponseCode(me.getResponseCode()));
+    } else {
+      resStatus.setErr(ResponseCode.SERVER_ERROR.name());
+      response.setResponseCode(ResponseCode.SERVER_ERROR);
+    }
+    response.setParams(resStatus);
+    return response;
   }
 
   /**
-   * Getter method for actor reference or selection object.
+   * This method will provide environment id.
    *
-   * @return Actor reference or selection object
+   * @return int
    */
-  public Object getActorRef() {
-    return actorRef;
+  private int getEnvironment() {
+
+    if (ServiceBaseGlobal.env != null) {
+      return ServiceBaseGlobal.env.getValue();
+    }
+    return ProjectUtil.Environment.dev.getValue();
+  }
+
+  private void generateTelemetryForSuccessResponse(Object response, Request request) {
```

Comment:
Don't require generateTelemetryForSuccessResponse and generateTelemetryForExceptionResponse.